packages/yarnpkg-core/tests/httpUtils.test.ts

```diff
@@ -0,0 +1,36 @@
+import {xfs}           from "@yarnpkg/fslib";
+
+import {Configuration} from "../sources/Configuration";
+
+import * as httpUtils  from "../sources/httpUtils";
+
+describe(`httpUtils`, () => {
+  it(`it should fail to make requests to a blocked hostname`, async () => {
```

Comment:
Can you rather implement this as an E2E test? Cf `enableNetwork.test.js`

(in general I tend to prefer E2E for anything that's exposed by the CLI, and keep unit tests to things that purely matter to the direct API consumers; the idea is that we should be able to tests our features against a completely different package manager implementation if we wanted to)

Reply:
Sure